packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -95,11 +106,84 @@ export async function forceSourceRetrieveSourcePath(explorerPath: vscode.Uri) {
       return;
     }
   }
+
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new FilePathGatherer(explorerPath),
-    new ForceSourceRetrieveSourcePathExecutor(),
+    useBetaRetrieve(explorerPath)
+      ? new LibraryRetrieveSourcePathExecutor()
+      : new ForceSourceRetrieveSourcePathExecutor(),
     new SourcePathChecker()
   );
   await commandlet.run();
 }
+
+// this supported types logic is temporary until we have a way of generating the metadata type from the path
+// once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+export function useBetaRetrieve(explorerPath: vscode.Uri): boolean {
+  const filePath = explorerPath.fsPath;
+  const betaDeployRetrieve = sfdxCoreSettings.getBetaDeployRetrieve();
+  const supportedType =
+    isAuraOrLWC(filePath) ||
+    path.extname(filePath) === APEX_CLASS_EXTENSION ||
+    filePath.includes(`${APEX_CLASS_EXTENSION}-meta.xml`) ||
+    (path.extname(filePath) === APEX_TRIGGER_EXTENSION ||
+      filePath.includes(`${APEX_TRIGGER_EXTENSION}-meta.xml`)) ||
+    (path.extname(filePath) === VISUALFORCE_COMPONENT_EXTENSION ||
+      filePath.includes(`${VISUALFORCE_COMPONENT_EXTENSION}-meta.xml`)) ||
+    (path.extname(filePath) === VISUALFORCE_PAGE_EXTENSION ||
+      filePath.includes(`${VISUALFORCE_PAGE_EXTENSION}-meta.xml`));
+  return betaDeployRetrieve && supportedType;
+}
+
+export function isAuraOrLWC(filePath: string) {
+  let res = false;
+  if (
+    !fs.statSync(filePath).isDirectory() ||
+    (!filePath.endsWith(AURA_DIRECTORY) && !filePath.endsWith(LWC_DIRECTORY))
+  ) {
+    const parent = path.dirname(filePath);
+    const grandParent = path.dirname(parent);
+    res =
+      parent.endsWith(AURA_DIRECTORY) ||
+      grandParent.endsWith(AURA_DIRECTORY) ||
```

Comment:
technically there could be x number of parents before reaching the component bundle itself (I believe this is the case but let me know otherwise) and it would be valid, so using the registry approach like i posted above might be more resilient